### ISSUE TITLE
SQS Queue and attached DLQ should have same type

### ISSUE THREAD

**Issue by abhinav-enboarder** [aws-cloudformation/cfn-lint#3950]
### CloudFormation Lint Version

0.79.7

### What operating system are you using?

MacOS

### Describe the bug

When creating an SQS queue of FIFO type, the attached DLQ to this should also be a FIFO one, the linter does not flag this as an error

### Expected behavior

When creating an SQS queue of FIFO type, the attached DLQ to this queue should be marked as invalid resource if it's not a FIFO queue

### Reproduction template

```yaml
Resources:
  MyDLQ:
    Type: AWS::SQS::Queue
    Properties:
      QueueName: MyDLQNameHere
      # FifoQueue: true
      ContentBasedDeduplication: true
      VisibilityTimeout: 1800
      MessageRetentionPeriod: 1209600

  MyQueue:
    Type: AWS::SQS::Queue
    Properties:
      QueueName: MyQueueNameHere.fifo
      FifoQueue: true
      ContentBasedDeduplication: true
      VisibilityTimeout: 1800
      MessageRetentionPeriod: 7200
      RedrivePolicy:
        deadLetterTargetArn: !GetAtt MyDLQ.Arn
        maxReceiveCount: 2

```

**Comment by kddejong** [comment]
Confirmed error message:

`Reason: Dead-letter queue must be same type of queue as the source.`

If `FifoQueue` is false (or not specified) and `ContentBasedDeduplication` is `true` then  `Unknown Attribute ContentBasedDeduplication.`

**Comment by abhinav-enboarder** [comment]
@kddejong I have tried this template with `v1.24.0` and still do  not get this error message

**Comment by kddejong** [comment]
Yep, agreed. This is a new rule request.

**Comment by kddejong** [comment]
Added a PR to do some validation on fifo/standard queue property validation and the association of the DLQ types.

**Comment by kddejong** [comment]
Release [v1.25.0](https://github.com/aws-cloudformation/cfn-lint/releases/tag/v1.25.0) has it.
